File: js/image_info.js
import { app } from "/scripts/app.js";

app.registerExtension({
    name: "nce.utils.image_info",
    async beforeRegisterNodeDef(nodeType, nodeData, app) {
        if (!nodeData?.category?.startsWith("🐍 NCE")) return;

        if (nodeData?.name === "NCEGetImageSize") {
            // 连接输入时设置初始标签
            const onConnectInput = nodeType.prototype.onConnectInput;
            nodeType.prototype.onConnectInput = function (targetSlot, type, output, originNode, originSlot) {
                const v = onConnectInput ? onConnectInput.apply(this, arguments) : undefined;
                this.outputs[1]["label"] = "width";
                this.outputs[2]["label"] = "height";
                this.outputs[3]["label"] = "count";
                return v;
            };

            // 执行后更新标签显示实际值
            const onExecuted = nodeType.prototype.onExecuted;
            nodeType.prototype.onExecuted = function (message) {
                const r = onExecuted ? onExecuted.apply(this, arguments) : undefined;

                if (message["text"]) {
                    let values = message["text"].toString().split('x').map(Number);
                    // 格式: count x width x height
                    this.outputs[1]["label"] = values[1] + " width";
                    this.outputs[2]["label"] = values[2] + " height";
                    this.outputs[3]["label"] = values[0] + " count";
                }

                return r;
            };
        }
    },
});

File: nodes/image_info.py
"""
图像信息节点
获取图像的尺寸和数量信息
"""

CATEGORY = "🐍 NCE/图像"


class NCEGetImageSize:
    """获取图像尺寸和数量"""
    
    @classmethod
    def INPUT_TYPES(s):
        return {
            "required": {
                "image": ("IMAGE",),
            }
        }

    RETURN_TYPES = ("IMAGE", "INT", "INT", "INT",)
    RETURN_NAMES = ("image", "width", "height", "count",)
    FUNCTION = "get_size"
    CATEGORY = CATEGORY
    DESCRIPTION = """
获取图像的宽度、高度和批次数量,
并将图像原样传递。
"""

    def get_size(self, image):
        width = image.shape[2]
        height = image.shape[1]
        count = image.shape[0]
        
        return {
            "ui": {
                "text": [f"{count}x{width}x{height}"]
            }, 
            "result": (image, width, height, count) 
        }
